web/proxy.js

```diff
@@ -1,21 +1,18 @@
 const http = require('http');
 const https = require('https');
+const proxyConfig = require('../config/proxyConfig');
 require('dotenv').config();
 
 if (process.env.USE_WEB_PROXY === 'false') {
     process.stdout.write('Skipping proxy as USE_WEB_PROXY was set to false.\n');
     process.exit();
 }
-
-let host = 'www.expensify.com';
-
-// If we are testing against the staging API then we must use the correct host here or nothing with work.
-if (/staging/.test(process.env.EXPENSIFY_URL)) {
-    host = 'staging.expensify.com';
-}
+const host = new URL(process.env.EXPENSIFY_URL || 'https://www.expensify.com').hostname;
+const stagingHost = new URL(process.env.STAGING_EXPENSIFY_URL || 'https://staging.expensify.com').hostname;
+const stagingSecureHost = new URL(process.env.STAGING_SECURE_EXPENSIFY_URL || 'https://staging-secure.expensify.com').hostname;
```

Comment:
So these two have been removed from the .env.example right? Would this be undefined or do you not want to include it in the example but in some instructions?

Reply:
@Prince-Mendiratta got it, but then who will use use these env variables? If contributors, will we have to update instructions to let them know, just missing  why its needed right now, thanks for bearing with me.

Reply:
> why anyone would ever modify the STAGING_EXPENSIFY_URL or EXPENSIFY_URL environment variables.

@Prince-Mendiratta Is this an existing use case though? I am just wondering if we should leave code in place which is not solving any problem right now, also there is no documentation about these env variables either if people will come across them and its not clear what they would use this for.

Reply:
Yeah, my take is that if there is no use for this now, let's not leave it in the codebase, super easy to add this back whne we actually be able to use it. Curious for other. 

Thanks for explaining 🙇

Reply:
I agree we can remove the `process.env` values - they aren't supposed to be set by external developers 

It can even simplify the expression down to
```suggestion
const stagingHost = 'staging.expensify.com';
const stagingSecureHost = 'https://staging-secure.expensify.com';
```

Reply:
Lets probably move those values to the Config if they are not there already

Reply:
Oh, sorry, thanks for clarifying!

Reply:
@kidroca @mountiny  Making [these changes](https://github.com/Expensify/App/pull/16432#discussion_r1147920466) will create some inconsistency.  Contributors will still be allowed to change api endpoints by changing .env in Desktop and native apps. It will just prevent use of .env in proxy(i.e, Web). Won't it create more confusion?

Reply:
Should we then change the instructions for the contributors to not change the env? They should be able to achieve what they need from the app no?

Reply:
So then it sounds like we could just update the instructions (if there are some, I am not sure actually)

Reply:
Oh, noise